extensions/sd_api_pictures/script.py

```diff
@@ -139,41 +188,76 @@ def bot_prefix_modifier(string):
 
     return string
 
-def force_pic():
-    global picture_response
-    picture_response = True
+def toggle_generation(*args):
+    global picture_response, shared, streaming_state
+    if not args:
+        picture_response = not picture_response
+    else:
+        picture_response = args[0]
+
+    shared.args.no_stream = True if picture_response else streaming_state # Disable streaming cause otherwise the SD-generated picture would return as a dud
+    shared.processing_message = "*Is sending a picture...*" if picture_response else "*Is typing...*"
+    btn_text = "Suppress the picture response" if picture_response else "Force the picture response"
+    
+    return btn_text
+
+def filter_address(address):
+    address = address.strip()
+    address = re.sub('http(s)?:\/\/|\/$','',address) # remove starting http:// OR https:// OR trailing slash
```

Comment:
Why is this done? This seems like the primary result is just going to be breaking functionality for anyone that uses `https`, and not making a difference for anyone that doesn't apply to.

Reply:
https://github.com/oobabooga/text-generation-webui/pull/309#issuecomment-1475828792
It seems that trying to connect to API via https breaks functionality for whoever uses it regardless

Reply:
That... seems more like user error in the config of the user that posted that. There's no reason use `https` on its own should cause issues -IF- https is configured properly aka not using a self-signed cert from an autowebui addon, and instead using a real https certifier (eg a cloudflare tunnel)

Reply:
Convincing. Reverted the https strip